src/cljs/org/broadinstitute/firecloud_ui/common/input.cljs

```diff
@@ -24,8 +24,11 @@
                                :style (merge (or (:style props) {})
                                         (when (:invalid @state)
                                           {:borderColor (:exception-red style/colors)}))
-                               :onChange #(swap! state dissoc :invalid)
+                               :onChange #(do (swap! state dissoc :invalid)
+                                              (when-let [x (:onChange props)]
+                                                (x (.. % -target -value))))
```

Comment:
May I ask for a walkthrough on each element of this line? I understand that x is the onChange handler, but that's it.